tests/feature_grammar.js

```diff
@@ -0,0 +1,40 @@
+var tape = require("tape");
+
+var protobuf = require("..");
+
+var protoEditionsRequired = `edition = "2023";
+    message A {\
+        required uint32 a = 1;\
+    }`;
+var protoEditionsOptional = `edition = "2023";
+    message A {\
+        optional uint32 a = 1;\
+    }`;
+
+var protoEditionsGroup = `edition = "2023";
+    message A {\
+        group uint32 a = 1;\
+    }`;
+
+var noQuote = `edition = "2023";
+   message Foo {
+        reserved bar, baz;
+   }`;
+
+tape.test("editions grammar", function(test) {
+    test.throws(function() {
+        protobuf.parse(protoEditionsRequired);
+    }, Error, "Error: illegal token 'required'");
+
+    test.throws(function() {
+        protobuf.parse(protoEditionsOptional);
+    }, Error, "Error: illegal token 'optional'");
+
+    test.throws(function() {
+        protobuf.parse(protoEditionsGroup);
+    }, Error, "Error: illegal token 'group'");
+
+    test.ok(protobuf.parse(noQuote));
+    
+    test.end();
+});
```

Comment:
I don't know how to split it up any further. What do you have in mind?

Reply:
e.g.
```
tape.test("editions required keyword", function(test) {
    test.throws(function() {
        protobuf.parse(`edition = "2023";
            message A {\
                required uint32 a = 1;\
            }`;);
    }, Error, "Error: illegal token 'required'");
});
```

I think ideally this would be 4 separate tests, since it's testing 4 distinct behaviors.

Reply:
Sure.